pkg/networkservice/mechanisms/vlan/client.go

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2021 Nordix Foundation.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vlan
+
+import (
+	"context"
+	"io"
+	"sync"
+	"time"
+
+	"git.fd.io/govpp.git/api"
+	interfaces "github.com/edwarnicke/govpp/binapi/interface"
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/cls"
+	vlanmech "github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/vlan"
+	"github.com/networkservicemesh/api/pkg/api/networkservice/payload"
+	kernellink "github.com/networkservicemesh/sdk-kernel/pkg/kernel"
+	"github.com/networkservicemesh/sdk-kernel/pkg/kernel/tools/nshandle"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/chain"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
+	"github.com/networkservicemesh/sdk/pkg/tools/postpone"
+
+	"github.com/networkservicemesh/sdk-vpp/pkg/networkservice/mechanisms/vlan/l2vtr"
+	"github.com/networkservicemesh/sdk-vpp/pkg/networkservice/mechanisms/vlan/linkinit"
+	"github.com/networkservicemesh/sdk-vpp/pkg/networkservice/mechanisms/vlan/pciaddresspool"
+	"github.com/networkservicemesh/sdk-vpp/pkg/tools/ifindex"
+)
+
+type vlanClient struct {
+	vppConn api.Connection
+}
+
+// NewClient returns a VLAN client chain element
+func NewClient(vppConn api.Connection, resourceLock sync.Locker, resourcePool pciaddresspool.PCIAddressPool) networkservice.NetworkServiceClient {
+	return chain.NewNetworkServiceClient(
+		l2vtr.NewClient(vppConn),
+		&vlanClient{vppConn: vppConn},
+		linkinit.NewClient(vppConn),
+		pciaddresspool.NewClient(resourceLock, resourcePool),
```

Comment:
I'm confused as to why pciaddresspool is here, and more so, why it's pulling in sdk-sriov into this PR...

Reply:
It's a simplified PCI address pool which maps the service domain to an interface name using the PCI addresses.

Reply:
Ah... OK... now I am remembering... and it pulls through sdk-sriov ?  Is there anyway to have it not do so simply?

Reply:
@ljkiraly If I am remembering correctly, the root issue here is "How to know which interface to bind to"  For tunnels we do this by passing a TunnelIP in as a parameter, and discovering which one to pass in in the cmd-forwarder-vpp.  What parameter might we pass in at the top here?

Reply:
No, the resourcepool in sdk-sriov is more complex and also a PCI pool is needed to get it working. To initialize the pools require additional parameters which are not used here. So the sdk-sriov is only used to process the configuration file. The pool and the mapping is in sdk-vpp.

Reply:
I moved the domain to device config to cmd-forwarder-vpp and pass the map to clients in sdk-vpp